### ISSUE TITLE
Long hashtags are broken

### ISSUE THREAD

**Issue by DrHyde** [mastodon/mastodon#25725]
### Steps to reproduce the problem

1. Make a toot with a very long hashtag. #Donau­dampf­schifffahrts­elektrizitäten­haupt­betriebs­werk­bau­unterbeamten­gesellschaft for example. I expect that the bug is actually triggered by something shorter than that, but that's what I used.
2. See that it gets broken into a small part that is a link and the rest is left as plain text. In this case (see screenshot) the first five characters are a link, to tag `#Donau` and the rest is plain text.

### Expected behaviour

The entire very long compound word should be a hashtag linking to #EntireVeryLongWord

### Actual behaviour

Only a small part of the string was made into a hashtag

### Detailed description

Note that the long compound word was too long to display on a single line when I was creating the toot, and is too long to display on a single line in the web interface. It got split at the point in the word where it had been automatically hyphenated by something - whether that something is the browser or some funky Javascript in Mastodon I don't know. When displayed later at a different font size the breakage still happens at the same point.

![Screenshot 2023-07-05 at 00 16 02](https://github.com/mastodon/mastodon/assets/216826/140b1aa6-497a-4e54-8963-5f9aa1304078)
![Screenshot 2023-07-05 at 00 22 23](https://github.com/mastodon/mastodon/assets/216826/281a6b3a-f499-4334-8a27-fd147e05bca6)

The specific toot where this happened is [here](https://fosstodon.org/@DrHyde/110658331623617447).

### Specifications

Mastodon 4.1.2, Chrome 114.0.5735.106, Mac OS 12.6.

**Comment by slayeh17** [comment]
@DrHyde I tried replicating this issue and it actually works for long hastags but only if it start with a lower case letter.

like [#donaudampfschifffahrtselektrizitatenhauptbetriebswerkbauunterbeamtengesellschaft](https://mastodon.social/tags/donaudampfschifffahrtselektrizitatenhauptbetriebswerkbauunterbeamtengesellschaft)

You can check my test [here](https://mastodon.social/@debak/110658920505443206).

Although the specific hastag [#Donau](https://mastodon.social/tags/Donau)­dampf­schifffahrts­elektrizitäten­haupt­betriebs­werk­bau­unterbeamten­gesellschaft seems to have some issue. Cause I also tried some hashtags which start with upper case letter, they seem to work.

**Comment by DrHyde** [comment]
Oh, it may also matter that while I'm using a German hashtag, my account is in English which may confuse matters regarding hyphenation

**Comment by slayeh17** [comment]
It's possible 🤷.

**Comment by kevintraver** [comment]
Yes, it looks like the hashtag in the referenced post contains an ASCII character 173 which is a non-printable soft-hypen.

**Comment by DrHyde** [comment]
For the avoidance of doubt ... I did not type a soft-hyphen when creating that toot :-)

**Comment by DrHyde** [comment]
I can no longer replicate the problem, so either it's been fixed in Mastodon or the ultimate cause was my browser being screwy and that has been fixed:

![Screenshot 2024-09-27 at 19 10 57](https://github.com/user-attachments/assets/5a981e4e-ff14-4827-88a6-0452d39e9c72)
![Screenshot 2024-09-27 at 19 11 10](https://github.com/user-attachments/assets/e43f8b17-aaa0-4c95-92cf-bc8bc1203075)
